ci/pipeline

```diff
@@ -273,9 +285,79 @@ def release(requestVersion: String, gitSha: String, latest: Boolean = false): Un
   // TODO: Publish native packages to unstable.
 }
 
+/**
+  * Given tagName v1.5.1, and rev DEADBEEF, we would assert that v1.5.0 is a direct ancestor of rev DEADBEEF
+  *
+  * Also, we assert that rev contains new commits since the last release. If the proposal is to tag v1.5.1 with same
+  * revision as v1.5.0, then we fail, saying that there v1.5.0 points to the commit porposed for v1.5.1
+  *
+  * @param tagName The tag name / version we are going to tag. eg v1.5.1
+  * @param gitSha The commit sha / rev that will be tagged.
+  */
+def assertPriorVersionIsAncestor(tagName: String, gitSha: String): Unit = {
+  utils.priorPatchVersion(tagName).foreach { prior =>
+    val revs = %%("git", "rev-list", "--format=oneline", s"${gitSha}..${prior}").out.string
+    if(revs.trim.nonEmpty) {
+      println(s"Error! ${prior} is not a direct ancestor of proposed commit ${gitSha} for ${tagName}!")
+      sys.exit(1)
+    }
+
+    val commitsSinceLast = %%("git", "rev-list", "--format=oneline", s"${prior}..${gitSha}").out.string
+    if(commitsSinceLast.trim.isEmpty) {
+      println(s"Error! ${prior} points to proposed commit ${gitSha} for ${tagName}!")
+      sys.exit(1)
+    }
+  }
+}
+
 def withoutVersionPrefix(releaseVersion: String): String = {
     if(releaseVersion.toLowerCase.startsWith("v"))
       releaseVersion.substring(1)
     else
       releaseVersion
 }
+
+def uploadLinuxPackagesToRepos(tagName: String): Unit = {
```

Comment:
This does not seem to be called anywhere. Should we upload each master build? As far as I understood we agreed to have the master builds on the unstable release channel.

Reply:
that may be true... I understand that if we do release from master that should be on the unstable release channel.  I was sure or clear if we agreed we should do that.   At this point I would really like to publish our releases.

Reply:
Ok, can we follow up then? Also, shouldn't this method be called in the release target?

Reply:
it is called in the release method... line 269